### DISCUSSION TITLE
HOGWARTS: Expel/Delete button functionality

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#76] [Help Me!] [answered]
Your Name
Have you searched the other help tickets? Yes, Rob's  #72 has helped me ,and I am trying to learn to use the spread operator.

Describe your issue 
--  My click event works, and I am trying to learn to use the spread, but I need help walking thru the thought process of declaring the different variables needed.

List any error messages that appear -- none

What have you googled? 
https://www.freecodecamp.org/news/lets-clear-up-the-confusion-around-the-slice-splice-split-methods-in-javascript-8ba3266c29ae/
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

What have you tried? 
I have created needed lines of code for push and slice methods, but I am unsure how to declare variables to use the spread method that Rob used.  

https://github.com/jim-conner/sorting-hat/blob/main/main.js

**Comment by jimguistical** [comment, marked as answer]
I needed to better understand what is happening when I used the event.target property to be able to use the methods.  Only then and what it actually looks like when you traverse one object to the next and then finally to type or id -- `e.target.id`  .  (see screen cap example attached).Also, I need console.log the events to understand them before I can use the array methods I want to use!
![Screenshot 2021-02-03 194147](https://user-images.githubusercontent.com/76793747/106833252-d44e7980-6658-11eb-8440-1377c1d58081.png)

I needed to console log deleteStudent to see how the array with nested object moved around.  Once I understood splice, then I could comprehend spread (...).  

```
const expelStudent = (e) => {
  const targetType = e.target.type;
  const targetId = e.target.id;
  // console.log(e);
  if (targetType === "button") {
    let deleteStudent = students.splice(targetId, 1);
    // console.log(deleteStudent)
    expelledArray.push(...deleteStudent);
  }
  
  // console.log(expelledArray);
  studentBuilder(students);
}
```

**Reply by drteresavasquez** [reply]
@jim-conner what would `expelledArray` look like if you did not use the spread on `deleteStudent` and you pushed it into the array.
